### DISCUSSION TITLE
Persistent code block IDs

### DISCUSSION THREAD

**Discussion by hellt** [facelessuser/pymdown-extensions#1959] [Ideas]
Hi, this stemmed from this discussion in mkdocs-material: https://github.com/squidfunk/mkdocs-material/discussions/5083#discussioncomment-5076157.

The gist of this idea is to allow code block extension to be provisioned with a custom `id` element, that downstream engines could leverage to make persistent ID elements used to refer to the code blocks in a predictable manner.

**Comment by facelessuser** [comment]
I think it is reasonable that we could add support for

````
``` {.language .class #id linenums=3}
some code
```
````

**Reply by facelessuser** [reply]
I've opened an issue based on this discussion.

**Reply by facelessuser** [reply]
@hellt  This is actually already supported, so there is nothing to do. I knew we supported additional classes, but as long as `attr_list` extensions enabled, you'll get classes and IDs using the notation above. Attributes on the other hand only support `data-` when using Pygments as all other `key=value` items are treated as options.

I will close the spawned issue as it is no longer relevant. I'll update documentation to make it more clear.

**Comment by facelessuser** [comment]
Just to update this conversation. Code blocks do allow specifying IDs, and now when things like line spans and line anchors are enabled for code blocks, that generate IDs as well, the IDs will utilize the main code block's ID to generate those related IDs.
